### PR TITLE
feat(ui5-stringreplace): create new module ui5-stringreplace

### PR DESCRIPTION
**Pull request by dominikfeininger**
merging:
- ui5-task-stringrepalce
- ui5-middleware-stringreplace

follow up from https://github.com/ui5-community/ui5-ecosystem-showcase/pull/675

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,18 +21,17 @@ packages
...
 ├── ui5-task-flatten-library        // task extension: prepares build result for deployment to SAP NetWeaver
 ├── ui5-task-i18ncheck              // task extension: checks for missing i18n texts
 ├── ui5-task-minify-xml             // task extension: minify xml resources
 ├── ui5-task-pwa-enabler            // task extension: enables ui5 app with pwa functionalities
-├── ui5-task-stringreplacer         // task extension: replaces placeholder strings
 ├── ui5-task-zipper                 // task extension: bundle the entire webapp in a zip-archive
 ├── ui5-tooling-modules             // tooling extensions: direct consumption of NPM packages in UI5 apps
-└── ui5-tooling-transpile           // tooling extensions: transpile resources using Babel
+├── ui5-tooling-transpile           // tooling extensions: transpile resources using Babel
+└── ui5-tooling-stringreplace      // tooling extension: replaces placeholder strings
```
Comment (petermuessig) [inline README.md:34]: `s` comes before `t` 😉

File: README.md [hunk 2]
```diff
@@ -128,13 +126,13 @@ The middlewares developed in this monorepo are also publicly available on NPM ri
...
 * https://www.npmjs.com/package/ui5-middleware-servestatic
 * https://www.npmjs.com/package/ui5-middleware-simpleproxy
-* https://www.npmjs.com/package/ui5-middleware-stringreplacer
 * https://www.npmjs.com/package/ui5-middleware-webjars
 
 The tooling extensions (contains tasks and middlewares) developed in this monorepo are available on NPM right here:
 
 * https://www.npmjs.com/package/ui5-tooling-modules
 * https://www.npmjs.com/package/ui5-tooling-transpile
+* https://www.npmjs.com/package/ui5-tooling-stringreplace
```
Comment (petermuessig) [inline README.md:135]: same here - alphabetic order

File: README.md [hunk 3]
```diff
@@ -172,6 +168,7 @@ Available tooling extensions in this project:
 | ----------- | ----------- | ----- |
 | [ui5-tooling-modules](packages/ui5-tooling-modules/README.md) | direct consumption of NPM packages | [![npm version](https://badge.fury.io/js/ui5-tooling-modules.svg)](https://badge.fury.io/js/ui5-tooling-modules) |
 | [ui5-tooling-transpile](packages/ui5-tooling-transpile/README.md) | transpile resources using Babel | [![npm version](https://badge.fury.io/js/ui5-tooling-transpile.svg)](https://badge.fury.io/js/ui5-tooling-transpile) |
+| [ui5-tooling-stringreplace](packages/ui5-tooling-stringreplace/README.md) | stringreplace resources using Babel | [![npm version](https://badge.fury.io/js/ui5-tooling-stringreplace.svg)](https://badge.fury.io/js/ui5-tooling-stringreplace) |
```
Comment (petermuessig) [inline README.md:171]: again - alphabetic order

File: packages/ui5-app/package.json [hunk 4]
```diff
@@ -72,15 +72,14 @@
...
-    "ui5-middleware-stringreplacer": "^0.5.1",
     "ui5-middleware-webjars": "^0.4.2",
     "ui5-task-cachebuster": "^0.3.2",
     "ui5-task-compileless": "^0.3.2",
     "ui5-task-i18ncheck": "^0.3.10",
     "ui5-task-minify-xml": "^0.3.0",
     "ui5-task-pwa-enabler": "^0.3.3",
-    "ui5-task-stringreplacer": "^0.8.0",
     "ui5-task-zipper": "^0.7.2",
+    "ui5-tooling-stringreplace": "^0.0.1",
```
Comment (petermuessig) [inline packages/ui5-app/package.json:82]: `s` comes after `m`

File: packages/ui5-app/package.json [hunk 5]
```diff
@@ -95,17 +94,16 @@
...
-      "ui5-middleware-stringreplacer",
       "ui5-middleware-webjars",
       "ui5-task-compileless",
       "ui5-task-minify-xml",
       "ui5-task-i18ncheck",
       "ui5-task-pwa-enabler",
-      "ui5-task-stringreplacer",
       "ui5-task-zipper",
       "ui5-task-cachebuster",
+      "ui5-tooling-stringreplace",
```
Comment (petermuessig) [inline packages/ui5-app/package.json:104]: like above

File: packages/ui5-tooling-stringreplace/lib/middleware.js [hunk 6]
```diff
@@ -1,6 +1,8 @@
 /* eslint-disable no-undef */
 const log = require("@ui5/logger").getLogger("server:custommiddleware:stringreplacer");
```
Comment (petermuessig) [inline packages/ui5-tooling-stringreplace/lib/middleware.js:2]: `stringreplacer` should renamed to `stringreplace`

File: packages/ui5-tooling-stringreplace/lib/task.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-unused-vars, no-undef */
+const log = require("@ui5/logger").getLogger("builder:customtask:stringreplacer");
```
Comment (petermuessig) [inline packages/ui5-tooling-stringreplace/lib/task.js:2]: `stringreplacer` should renamed to `stringreplace`

File: packages/ui5-tooling-stringreplace/lib/task.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+ * @param {object} parameters.options Options
+ * @param {Array} parameters.options.files all file name patterns where replace should occur
+ * @param {Array} [parameters.options.strings] Array of objects containing placeholder and replacment text value
+ * @returns {Promise<undefined>} Promise resolving with undefined once data has been written
+ */
+module.exports = function ({ workspace, options }) {
+	const isDebug = options.configuration && options.configuration.debug;
+
+	// extract the placeholder strings from the configuration
+	if (options.configuration) {
```
Comment (petermuessig) [inline packages/ui5-tooling-stringreplace/lib/task.js:24]: Can be simplified today like this:

```
option.configuration?.replace?.forEach((entry) => {
```

**Review by petermuessig — CHANGES_REQUESTED** [review]
Besides some small findings, it looks good!

Once fixed, we can merge => 👍 

THX, Dominik

**Comment by petermuessig** [comment]
@dominikfeininger : Maybe you can also take a look into #339 and include this into the change?

**Comment by dominikfeininger** [comment]
Hi, I noticed there is neither an execution of the `ui5-tooling-stringreplace-middleware` nor `ui5-tooling-stringreplace-task` in the executed pipeline tests. At a first glance this seems to apply for other modules as well. IMHO each module should be executed at least once during the pipeline tests.

**Comment by dominikfeininger** [comment]
- merged:
 ui5-task-stringrepalce
 ui5-middleware-stringreplace
- add custom `prefix` option in ui5.yaml configuration

**Review by petermuessig — APPROVED** [review]
LGTM - thanks for the adoption...

**Pull request merged**